plugin_manager.py

```diff
@@ -13,40 +13,69 @@ class PluginManager:
+        self.loaded_filters = []
+        self.blocked = {}  # Dictionary to prevent multiple additions of same plugin/filter to queue
         self.config = config_loader.load_config('twilight')
 
+    def get_files(self, location):
+        """Helper function to retrieve plugin and filter scripts"""
+        files = []
+        for plugin_folder in os.listdir(location):
+            plugin_files = os.listdir(location+plugin_folder)
+            if "__init__.py" in plugin_files:
+                files.append(plugin_folder)
+        return files
+
     def get_available_plugins(self):
         """List all plugins located in the folder given in the config PLUGINS_FOLDER"""
         available_plugins = {}
         location = "." + self.config["PLUGINS_FOLDER"]
-        for plugin_folder in os.listdir(location):
-            plugin_files = os.listdir(location+plugin_folder)
-            if "__init__.py" in plugin_files:
-                try:
-                    available_plugins[plugin_folder] = {
-                        "name": plugin_folder,
-                        "path": location+plugin_folder+"/__init__.py",
-                        "priority": self.plugins["PLUGINS"][plugin_folder]['priority'],
-                        "persistent": self.plugins["PLUGINS"][plugin_folder]['persistent']
-                    }
-                except KeyError:
-                    # config.json not correctly formatted
-                    pass
+        for plugin_folder in self.get_files(location):
+            if plugin_folder in self.plugins["PLUGINS"]:
+                available_plugins[plugin_folder] = {
+                    "name": plugin_folder,
+                    "path": location+plugin_folder+"/__init__.py",
+                    "priority": self.plugins["PLUGINS"][plugin_folder]['priority'],
+                    "persistent": self.plugins["PLUGINS"][plugin_folder]['persistent']
+                }
             else:
-                # incorrectly formatted plugin
-                pass
+                available_plugins[plugin_folder] = {
+                    "name": plugin_folder,
+                    "path": location+plugin_folder+"/__init__.py",
+                    "priority": 0,
+                    "persistent": False
+                }
         return available_plugins
 
+    def get_available_filters(self):
+        """List all filters located in the folder given in the config FILTERS_FOLDER"""
+        available_filters = {}
+        location = "." + self.config["FILTERS_FOLDER"]
+        for filter_folder in self.get_files(location):
+            if filter_folder in self.plugins["FILTERS"]:
+                available_filters[filter_folder] = {
+                    "name": filter_folder,
+                    "path": location+filter_folder+"/__init__.py",
+                    "priority": self.plugins["FILTERS"][filter_folder]["priority"]
+                }
+            else:
+                available_filters[filter_folder] = {
```

Comment:
Remove this failsafe and throw an exception instead please.
Ping me if this seems like an odd request 😄